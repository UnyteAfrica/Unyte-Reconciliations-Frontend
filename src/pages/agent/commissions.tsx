import { Commission } from "@/types/types";
import { BiChevronDown } from "react-icons/bi";
import { useState } from "react";
import { cx } from "class-variance-authority";
import { WithLabel } from "@/components/shared/with-label";
import { CustomInput, DateInput } from "@/components/shared/input";
import { nairaSign } from "@/utils/utils";
import { Selector } from "@/components/shared/selector";
import { Pagination } from "@/components/shared/pagination";
import { CommissionsTable } from "@/components/tables/commissions-table";
import { SearchBar } from "@/components/searchbar";
import { DownloadButton } from "@/components/download-button";

const commissions: Commission[] = [
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Comprehensive",
    date: "May 7, 2023",
    commission: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Credit Life",
    date: "May 7, 2023",
    commission: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Card Protect",
    date: "May 7, 2023",
    commission: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Student",
    date: "May 7, 2023",
    commission: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Travel",
    date: "May 7, 2023",
    commission: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Device",
    date: "May 7, 2023",
    commission: 40000,
  },
];

export const periods = ["Daily", "Weekly", "Monthly", "Yearly"] as const;
export const products = [
  "Comprehensive",
  "Credit Life",
  "Car Protect",
  "Student",
  "Travel",
  "Device",
] as const;

export const Commissions = () => {
  const [isFilterOpen, setIsFilterOpen] = useState(false);
  const [period, setPeriod] = useState<string>(periods[0]);
  const [product, setProduct] = useState<string>(products[0]);
  const [page, setPage] = useState(1);
  const [startDate, setStartDate] = useState(new Date(1699885840400));
  const [endDate, setEndDate] = useState(new Date(1699885870400));

  const handleSearch = () => {
    //
  };

  return (
    <>
      <div
        className="mx-auto max-w-6xl mt-12 px-6 lg:px-0"
        onClick={() => setIsFilterOpen(false)}
      >
        <div className="flex flex-row justify-between items-center mb-10 relative">
          <span className="text-3xl text-[#333333] font-semibold">
            Commissions
          </span>
          <SearchBar
            handleSearch={handleSearch}
            placeholder="Find policy reference"
          />
          <div id="dates" className="flex flex-row items-center space-x-3">
            <button
              className="text-[#333333]"
              onClick={(e) => {
                e.stopPropagation();
                setIsFilterOpen((isFilterOpen) => !isFilterOpen);
              }}
            >
              <div className="space-x-2 flex flex-row items-center bg-[#828282] px-4 py-2 rounded text-white">
                <span className="text-base">Filter</span>
                <BiChevronDown
                  className={cx(
                    "transition-all duration-300",
                    isFilterOpen && "rotate-180"
                  )}
                />
              </div>
            </button>
            <DownloadButton data={[]} />
          </div>
          <div
            data-testid="filter"
            className={cx(
              "bg-white px-5 py-10 absolute top-[50px] w-[450px] right-0 rounded-lg border border-[#ccc] space-y-6 transition-all overflow-hidden opacity-0",
              !isFilterOpen && "max-h-0",
              isFilterOpen && "max-h-[500px] opacity-100 overflow-y-auto"
            )}
          >
            <WithLabel label="Period Range">
              <Selector
                options={periods}
                containerClassName="w-full"
                value={period}
                onChange={(val) => setPeriod(val)}
              />
            </WithLabel>
            <WithLabel label="Custom Date Range">
              <div className="flex w-full">
                <DateInput
                  containerClassName="rounded-tr-none rounded-br-none font-semibold w-1/2"
                  date={startDate}
                  onDateChange={(date) => setStartDate(date)}
                />
                <DateInput
                  containerClassName="rounded-tl-none rounded-bl-none font-semibold w-1/2"
                  date={endDate}
                  onDateChange={(date) => setEndDate(date)}
                />
              </div>
            </WithLabel>
            <CustomInput
              className="px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold"
              label="Policy No."
              labelClassName="text-base"
              placeholder="#A023457"
            />
            <WithLabel label="Select Product(s)">
              <Selector
                options={products}
                containerClassName="w-full"
                value={product}
                onChange={(val) => setProduct(val)}
              />
            </WithLabel>
            <CustomInput
              className="px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold"
              label="Amount"
              labelClassName="text-base"
              placeholder={nairaSign}
            />
            <div className="flex ml-auto">
              <button
                className="bg-white border rounded-md w-[90px] inline-block mr-4 py-2 ml-auto"
                onClick={() => setIsFilterOpen(false)}
              >
                Cancel
              </button>
              <button className="bg-primary text-white w-[115px] rounded-md py-2">
                Apply Filter
              </button>
            </div>
          </div>
        </div>
        <div className="mb-10">
          <CommissionsTable commissions={commissions} />
        </div>
        <Pagination
          currentPage={page}
          itemsCount={56}
          pageSize={10}
          onPageChange={(page) => setPage(page)}
        />
      </div>
    </>
  );
};
