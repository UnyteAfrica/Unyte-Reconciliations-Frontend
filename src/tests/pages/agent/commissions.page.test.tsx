import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import React from "react";
import { Commissions } from "@/pages/agent/commissions";

function setup(reactNode: React.ReactNode) {
  return {
    user: userEvent.setup(),
    ...render(reactNode),
  };
}

describe("Agent Policies Page", () => {
  it("renders correctly", () => {
    const tree = render(
      <BrowserRouter>
        <Commissions />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it("should show filter menu when filter is clicked", async () => {
    const { user } = setup(
      <BrowserRouter>
        <Commissions />
      </BrowserRouter>
    );

    const filterBtn = screen.getByText("Filter");
    await user.click(filterBtn);
    const filterBox = screen.getByTestId("filter");

    expect(filterBox).toHaveClass("max-h-[500px]");
    expect(filterBox).not.toHaveClass("max-h-0");
  });

  it("should close filter menu when it the user clicks outside it", async () => {
    const { user } = setup(
      <BrowserRouter>
        <Commissions />
      </BrowserRouter>
    );

    const filterBtn = screen.getByText("Filter");
    await user.click(filterBtn);

    const commissionsText = screen.getByText("Commissions");
    await user.click(commissionsText);
    const filterBox = screen.getByTestId("filter");

    expect(filterBox).toHaveClass("max-h-0");
    expect(filterBox).not.toHaveClass("max-h-[500px]");
  });
});
