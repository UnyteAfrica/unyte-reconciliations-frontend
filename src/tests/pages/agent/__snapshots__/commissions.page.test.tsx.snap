// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Agent Commissions Page > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto max-w-6xl mt-12 px-6 mb-16 lg:px-0"
      >
        <div
          class="flex flex-row justify-between items-center mb-10 relative"
        >
          <span
            class="text-3xl text-[#333333] font-semibold capitalize"
            data-testid="pageTitle"
          >
            Commissions
          </span>
          <form>
            <div
              class="relative"
            >
              <input
                class="border rounded bg-[#F2F2F2] pl-9 px-6 py-2 placeholder:text-[#828282] text-opacity-20"
                placeholder="Find policy reference"
                type="search"
                value=""
              />
              <img
                alt="search icon"
                class="absolute top-3.5 left-2"
                src="/src/assets/Icons/SearchIcon.svg"
              />
            </div>
          </form>
          <div
            class="flex flex-row items-center space-x-3"
            id="dates"
          >
            <button
              class="text-[#333333]"
            >
              <div
                class="space-x-2 flex flex-row items-center bg-[#828282] px-4 py-2 rounded text-white"
              >
                <span
                  class="text-base"
                >
                  Filter
                </span>
                <svg
                  class="transition-all duration-300"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                  />
                </svg>
              </div>
            </button>
            <button
              class="bg-[#25D366] rounded"
            >
              <div
                class="space-x-2 flex flex-row items-center px-4 py-2 text-white"
              >
                <span
                  class="text-base"
                >
                  Download
                </span>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m12 16 4-5h-3V4h-2v7H8z"
                  />
                  <path
                    d="M20 18H4v-7H2v7c0 1.103.897 2 2 2h16c1.103 0 2-.897 2-2v-7h-2v7z"
                  />
                </svg>
              </div>
            </button>
          </div>
          <div
            class="bg-white px-5 py-10 absolute top-[50px] w-[450px] right-0 rounded-lg border border-[#ccc] space-y-6 transition-all overflow-hidden opacity-0 max-h-0"
            data-testid="filter"
          >
            <div>
              <label
                class="mb-2 inline-block"
              >
                Period Range
              </label>
              <div
                class="relative w-full"
              >
                <button
                  class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
                >
                  <div
                    class="space-x-2 flex flex-row items-center justify-between"
                  >
                    <span
                      class="text-base"
                      title="selectBtn"
                    >
                      Daily
                    </span>
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                      />
                    </svg>
                  </div>
                </button>
                <div
                  class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
                >
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Daily
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Weekly
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Monthly
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Yearly
                  </button>
                  <hr />
                </div>
              </div>
            </div>
            <div>
              <label
                class="mb-2 inline-block"
              >
                Custom Date Range
              </label>
              <div
                class="flex w-full"
              >
                <button
                  class="px-4 py-2 border rounded-md relative rounded-tr-none rounded-br-none font-semibold w-1/2"
                >
                  November 13, 2023
                  <input
                    class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                    placeholder="date"
                    type="date"
                    value="2023-11-13"
                  />
                </button>
                <button
                  class="px-4 py-2 border rounded-md relative rounded-tl-none rounded-bl-none font-semibold w-1/2"
                >
                  November 13, 2023
                  <input
                    class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                    placeholder="date"
                    type="date"
                    value="2023-11-13"
                  />
                </button>
              </div>
            </div>
            <div
              class=""
            >
              <label
                class="text-[#333] mb-2 block text-base"
                for="policyNo"
              >
                Policy No.
                 
              </label>
              <input
                class="block font-sora w-full border rounded-lg outline-none focus:border-black px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold"
                id="policyNo"
                placeholder="#A023457"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div>
              <label
                class="mb-2 inline-block"
              >
                Select Product(s)
              </label>
              <div
                class="relative w-full"
              >
                <button
                  class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
                >
                  <div
                    class="space-x-2 flex flex-row items-center justify-between"
                  >
                    <span
                      class="text-base"
                      title="selectBtn"
                    >
                      Comprehensive
                    </span>
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                      />
                    </svg>
                  </div>
                </button>
                <div
                  class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
                >
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Comprehensive
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Credit Life
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Card Protect
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Student
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Travel
                  </button>
                  <hr />
                  <button
                    class="w-full font-inter py-4 font-semibold"
                  >
                    Device
                  </button>
                  <hr />
                </div>
              </div>
            </div>
            <div
              class=""
            >
              <label
                class="text-[#333] mb-2 block text-base"
                for="amount"
              >
                Amount
                 
              </label>
              <input
                class="block font-sora w-full border rounded-lg outline-none focus:border-black px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold"
                id="amount"
                placeholder="₦"
              />
              <p
                class="text-sm text-[#333]"
              />
            </div>
            <div
              class="flex ml-auto"
            >
              <button
                class="bg-white border rounded-md w-[90px] inline-block mr-4 py-2 ml-auto"
              >
                Cancel
              </button>
              <button
                class="bg-primary text-white w-[115px] rounded-md py-2"
              >
                Apply Filter
              </button>
            </div>
          </div>
        </div>
        <div
          class="mb-10"
        >
          <table
            class="table-auto w-full"
          >
            <thead
              class="bg-gray-200 text-[#333333]"
            >
              <tr>
                <th
                  class="p-4 text-center"
                >
                  Policy Ref.
                </th>
                <th
                  class="p-4 text-center"
                >
                  Policy No.
                </th>
                <th
                  class="p-4 text-center"
                >
                  Product
                </th>
                <th
                  class="p-4 text-center"
                >
                  Date
                </th>
                <th
                  class="p-4 text-center"
                >
                  Commission
                </th>
              </tr>
            </thead>
            <tbody
              class="bg-white text-[#333333]"
            >
              <tr
                class="border-b font-medium"
              >
                <td
                  class="p-4 text-center"
                >
                  #WP62F3E8F93
                </td>
                <td
                  class="p-4 text-center"
                >
                  123jkf5402
                </td>
                <td
                  class="p-4 text-center"
                >
                  Comprehensive
                </td>
                <td
                  class="p-4 text-center"
                >
                  May 7, 2023
                </td>
                <td
                  class="p-4 text-center"
                >
                  ₦40,000
                </td>
              </tr>
              <tr
                class="border-b font-medium"
              >
                <td
                  class="p-4 text-center"
                >
                  #WP62F3E8F93
                </td>
                <td
                  class="p-4 text-center"
                >
                  123jkf5402
                </td>
                <td
                  class="p-4 text-center"
                >
                  Credit Life
                </td>
                <td
                  class="p-4 text-center"
                >
                  May 7, 2023
                </td>
                <td
                  class="p-4 text-center"
                >
                  ₦40,000
                </td>
              </tr>
              <tr
                class="border-b font-medium"
              >
                <td
                  class="p-4 text-center"
                >
                  #WP62F3E8F93
                </td>
                <td
                  class="p-4 text-center"
                >
                  123jkf5402
                </td>
                <td
                  class="p-4 text-center"
                >
                  Card Protect
                </td>
                <td
                  class="p-4 text-center"
                >
                  May 7, 2023
                </td>
                <td
                  class="p-4 text-center"
                >
                  ₦40,000
                </td>
              </tr>
              <tr
                class="border-b font-medium"
              >
                <td
                  class="p-4 text-center"
                >
                  #WP62F3E8F93
                </td>
                <td
                  class="p-4 text-center"
                >
                  123jkf5402
                </td>
                <td
                  class="p-4 text-center"
                >
                  Student
                </td>
                <td
                  class="p-4 text-center"
                >
                  May 7, 2023
                </td>
                <td
                  class="p-4 text-center"
                >
                  ₦40,000
                </td>
              </tr>
              <tr
                class="border-b font-medium"
              >
                <td
                  class="p-4 text-center"
                >
                  #WP62F3E8F93
                </td>
                <td
                  class="p-4 text-center"
                >
                  123jkf5402
                </td>
                <td
                  class="p-4 text-center"
                >
                  Travel
                </td>
                <td
                  class="p-4 text-center"
                >
                  May 7, 2023
                </td>
                <td
                  class="p-4 text-center"
                >
                  ₦40,000
                </td>
              </tr>
              <tr
                class="border-b font-medium"
              >
                <td
                  class="p-4 text-center"
                >
                  #WP62F3E8F93
                </td>
                <td
                  class="p-4 text-center"
                >
                  123jkf5402
                </td>
                <td
                  class="p-4 text-center"
                >
                  Device
                </td>
                <td
                  class="p-4 text-center"
                >
                  May 7, 2023
                </td>
                <td
                  class="p-4 text-center"
                >
                  ₦40,000
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <nav
          class="flex space-x-4"
        >
          <button
            class="px-5 py-2 rounded-md border text-sm font-medium text-[#ccc]"
            disabled=""
          >
            Previous
          </button>
          <ul
            class="flex space-x-2"
          >
            <li
              class="border rounded-md bg-primary text-white border-none"
              data-testid="active"
            >
              <button
                class="h-[38px] w-[38px]"
              >
                1
              </button>
            </li>
            <li
              class="border rounded-md"
              data-testid="false"
            >
              <button
                class="h-[38px] w-[38px]"
              >
                2
              </button>
            </li>
            <li
              class="border rounded-md"
              data-testid="false"
            >
              <button
                class="h-[38px] w-[38px]"
              >
                3
              </button>
            </li>
            <li
              class="border rounded-md"
              data-testid="false"
            >
              <button
                class="h-[38px] w-[38px]"
              >
                4
              </button>
            </li>
            <li
              class="border rounded-md"
              data-testid="false"
            >
              <button
                class="h-[38px] w-[38px]"
              >
                5
              </button>
            </li>
            <li
              class="border rounded-md"
              data-testid="false"
            >
              <button
                class="h-[38px] w-[38px]"
              >
                6
              </button>
            </li>
          </ul>
          <button
            class="px-5 py-2 rounded-md border text-sm font-medium"
          >
            Next
          </button>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto max-w-6xl mt-12 px-6 mb-16 lg:px-0"
    >
      <div
        class="flex flex-row justify-between items-center mb-10 relative"
      >
        <span
          class="text-3xl text-[#333333] font-semibold capitalize"
          data-testid="pageTitle"
        >
          Commissions
        </span>
        <form>
          <div
            class="relative"
          >
            <input
              class="border rounded bg-[#F2F2F2] pl-9 px-6 py-2 placeholder:text-[#828282] text-opacity-20"
              placeholder="Find policy reference"
              type="search"
              value=""
            />
            <img
              alt="search icon"
              class="absolute top-3.5 left-2"
              src="/src/assets/Icons/SearchIcon.svg"
            />
          </div>
        </form>
        <div
          class="flex flex-row items-center space-x-3"
          id="dates"
        >
          <button
            class="text-[#333333]"
          >
            <div
              class="space-x-2 flex flex-row items-center bg-[#828282] px-4 py-2 rounded text-white"
            >
              <span
                class="text-base"
              >
                Filter
              </span>
              <svg
                class="transition-all duration-300"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                />
              </svg>
            </div>
          </button>
          <button
            class="bg-[#25D366] rounded"
          >
            <div
              class="space-x-2 flex flex-row items-center px-4 py-2 text-white"
            >
              <span
                class="text-base"
              >
                Download
              </span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m12 16 4-5h-3V4h-2v7H8z"
                />
                <path
                  d="M20 18H4v-7H2v7c0 1.103.897 2 2 2h16c1.103 0 2-.897 2-2v-7h-2v7z"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="bg-white px-5 py-10 absolute top-[50px] w-[450px] right-0 rounded-lg border border-[#ccc] space-y-6 transition-all overflow-hidden opacity-0 max-h-0"
          data-testid="filter"
        >
          <div>
            <label
              class="mb-2 inline-block"
            >
              Period Range
            </label>
            <div
              class="relative w-full"
            >
              <button
                class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
              >
                <div
                  class="space-x-2 flex flex-row items-center justify-between"
                >
                  <span
                    class="text-base"
                    title="selectBtn"
                  >
                    Daily
                  </span>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                    />
                  </svg>
                </div>
              </button>
              <div
                class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
              >
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Daily
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Weekly
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Monthly
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Yearly
                </button>
                <hr />
              </div>
            </div>
          </div>
          <div>
            <label
              class="mb-2 inline-block"
            >
              Custom Date Range
            </label>
            <div
              class="flex w-full"
            >
              <button
                class="px-4 py-2 border rounded-md relative rounded-tr-none rounded-br-none font-semibold w-1/2"
              >
                November 13, 2023
                <input
                  class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                  placeholder="date"
                  type="date"
                  value="2023-11-13"
                />
              </button>
              <button
                class="px-4 py-2 border rounded-md relative rounded-tl-none rounded-bl-none font-semibold w-1/2"
              >
                November 13, 2023
                <input
                  class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                  placeholder="date"
                  type="date"
                  value="2023-11-13"
                />
              </button>
            </div>
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-base"
              for="policyNo"
            >
              Policy No.
               
            </label>
            <input
              class="block font-sora w-full border rounded-lg outline-none focus:border-black px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold"
              id="policyNo"
              placeholder="#A023457"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div>
            <label
              class="mb-2 inline-block"
            >
              Select Product(s)
            </label>
            <div
              class="relative w-full"
            >
              <button
                class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
              >
                <div
                  class="space-x-2 flex flex-row items-center justify-between"
                >
                  <span
                    class="text-base"
                    title="selectBtn"
                  >
                    Comprehensive
                  </span>
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                    />
                  </svg>
                </div>
              </button>
              <div
                class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
              >
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Comprehensive
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Credit Life
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Card Protect
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Student
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Travel
                </button>
                <hr />
                <button
                  class="w-full font-inter py-4 font-semibold"
                >
                  Device
                </button>
                <hr />
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-base"
              for="amount"
            >
              Amount
               
            </label>
            <input
              class="block font-sora w-full border rounded-lg outline-none focus:border-black px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold"
              id="amount"
              placeholder="₦"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class="flex ml-auto"
          >
            <button
              class="bg-white border rounded-md w-[90px] inline-block mr-4 py-2 ml-auto"
            >
              Cancel
            </button>
            <button
              class="bg-primary text-white w-[115px] rounded-md py-2"
            >
              Apply Filter
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-10"
      >
        <table
          class="table-auto w-full"
        >
          <thead
            class="bg-gray-200 text-[#333333]"
          >
            <tr>
              <th
                class="p-4 text-center"
              >
                Policy Ref.
              </th>
              <th
                class="p-4 text-center"
              >
                Policy No.
              </th>
              <th
                class="p-4 text-center"
              >
                Product
              </th>
              <th
                class="p-4 text-center"
              >
                Date
              </th>
              <th
                class="p-4 text-center"
              >
                Commission
              </th>
            </tr>
          </thead>
          <tbody
            class="bg-white text-[#333333]"
          >
            <tr
              class="border-b font-medium"
            >
              <td
                class="p-4 text-center"
              >
                #WP62F3E8F93
              </td>
              <td
                class="p-4 text-center"
              >
                123jkf5402
              </td>
              <td
                class="p-4 text-center"
              >
                Comprehensive
              </td>
              <td
                class="p-4 text-center"
              >
                May 7, 2023
              </td>
              <td
                class="p-4 text-center"
              >
                ₦40,000
              </td>
            </tr>
            <tr
              class="border-b font-medium"
            >
              <td
                class="p-4 text-center"
              >
                #WP62F3E8F93
              </td>
              <td
                class="p-4 text-center"
              >
                123jkf5402
              </td>
              <td
                class="p-4 text-center"
              >
                Credit Life
              </td>
              <td
                class="p-4 text-center"
              >
                May 7, 2023
              </td>
              <td
                class="p-4 text-center"
              >
                ₦40,000
              </td>
            </tr>
            <tr
              class="border-b font-medium"
            >
              <td
                class="p-4 text-center"
              >
                #WP62F3E8F93
              </td>
              <td
                class="p-4 text-center"
              >
                123jkf5402
              </td>
              <td
                class="p-4 text-center"
              >
                Card Protect
              </td>
              <td
                class="p-4 text-center"
              >
                May 7, 2023
              </td>
              <td
                class="p-4 text-center"
              >
                ₦40,000
              </td>
            </tr>
            <tr
              class="border-b font-medium"
            >
              <td
                class="p-4 text-center"
              >
                #WP62F3E8F93
              </td>
              <td
                class="p-4 text-center"
              >
                123jkf5402
              </td>
              <td
                class="p-4 text-center"
              >
                Student
              </td>
              <td
                class="p-4 text-center"
              >
                May 7, 2023
              </td>
              <td
                class="p-4 text-center"
              >
                ₦40,000
              </td>
            </tr>
            <tr
              class="border-b font-medium"
            >
              <td
                class="p-4 text-center"
              >
                #WP62F3E8F93
              </td>
              <td
                class="p-4 text-center"
              >
                123jkf5402
              </td>
              <td
                class="p-4 text-center"
              >
                Travel
              </td>
              <td
                class="p-4 text-center"
              >
                May 7, 2023
              </td>
              <td
                class="p-4 text-center"
              >
                ₦40,000
              </td>
            </tr>
            <tr
              class="border-b font-medium"
            >
              <td
                class="p-4 text-center"
              >
                #WP62F3E8F93
              </td>
              <td
                class="p-4 text-center"
              >
                123jkf5402
              </td>
              <td
                class="p-4 text-center"
              >
                Device
              </td>
              <td
                class="p-4 text-center"
              >
                May 7, 2023
              </td>
              <td
                class="p-4 text-center"
              >
                ₦40,000
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <nav
        class="flex space-x-4"
      >
        <button
          class="px-5 py-2 rounded-md border text-sm font-medium text-[#ccc]"
          disabled=""
        >
          Previous
        </button>
        <ul
          class="flex space-x-2"
        >
          <li
            class="border rounded-md bg-primary text-white border-none"
            data-testid="active"
          >
            <button
              class="h-[38px] w-[38px]"
            >
              1
            </button>
          </li>
          <li
            class="border rounded-md"
            data-testid="false"
          >
            <button
              class="h-[38px] w-[38px]"
            >
              2
            </button>
          </li>
          <li
            class="border rounded-md"
            data-testid="false"
          >
            <button
              class="h-[38px] w-[38px]"
            >
              3
            </button>
          </li>
          <li
            class="border rounded-md"
            data-testid="false"
          >
            <button
              class="h-[38px] w-[38px]"
            >
              4
            </button>
          </li>
          <li
            class="border rounded-md"
            data-testid="false"
          >
            <button
              class="h-[38px] w-[38px]"
            >
              5
            </button>
          </li>
          <li
            class="border rounded-md"
            data-testid="false"
          >
            <button
              class="h-[38px] w-[38px]"
            >
              6
            </button>
          </li>
        </ul>
        <button
          class="px-5 py-2 rounded-md border text-sm font-medium"
        >
          Next
        </button>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
